annotations/views/project_views.py

```diff
@@ -8,60 +8,61 @@
 from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
 from django.core.exceptions import PermissionDenied
 from django.http import HttpResponse, HttpResponseRedirect
-from django.contrib.auth import login, authenticate
 from django.conf import settings
-from django.db.models import Q, Count
+from django.db.models import Q
 
-from annotations.models import TextCollection, RelationSet
+from annotations.models import TextCollection, VogonUser
 from annotations.forms import ProjectForm
 
+from django.contrib import messages
+from django.http import Http404
+
+from annotations.utils import get_ordering_metadata, get_user_project_stats, _annotate_project_counts
+
 @login_required
 def view_project(request, project_id):
     """
     Shows details about a specific project owned by the current user.
-
-    Parameters
-    ----------
-    request : `django.http.requests.HttpRequest`
-    project_id : int
-
-    Returns
-    ----------
-    :class:`django.http.response.HttpResponse`
     """
+    project = get_object_or_404(
+        _annotate_project_counts(TextCollection.objects),
+        pk=project_id
+    )
+
+    # Check if user is owner or collaborator
+    if not (request.user == project.ownedBy or request.user in project.collaborators.all()):
+        raise PermissionDenied("Oops! Looks like you're trying to sneak a peek into someone else's project.")
 
-    project = get_object_or_404(TextCollection, pk=project_id)
     template = "annotations/project_details.html"
 
-    order_by = request.GET.get('order_by', 'title')
-    texts = project.texts.all().order_by(order_by)\
-                         .values('id', 'title', 'added', 'repository_source_id')
+    # Get ordering metadata
+    ordering = get_ordering_metadata(request, default_field='title', allowed_fields=['title', 'added'])
+
+    # Apply ordering
+    texts = project.texts.all().order_by(ordering['order_param'])\
```

Comment:
same as in other place, we don't want to list all texts in a proejct on one page (what if it's multiple hundreds?).